source/extensions/filters/http/adaptive_concurrency/adaptive_concurrency_filter.h

```diff
@@ -28,15 +28,29 @@ class AdaptiveConcurrencyFilterConfig {
 public:
   AdaptiveConcurrencyFilterConfig(
       const envoy::config::filter::http::adaptive_concurrency::v2alpha::AdaptiveConcurrency&
-          adaptive_concurrency,
+          proto_config,
       Runtime::Loader& runtime, std::string stats_prefix, Stats::Scope& scope,
       TimeSource& time_source);
 
+  // Returns whether the filter is enabled or should operate as a pass-through for a request.
+  bool filterEnabled() const {
+    // RuntimeFeatureFlag proto validates that the runtime key must have at least a single
+    // character. If the key is empty, the field was never specified.
+    if (enabled_runtime_key_.empty()) {
+      return true;
+    }
+
+    return runtime_.snapshot().getBoolean(enabled_runtime_key_, enabled_default_value_);
+  }
+
   TimeSource& timeSource() const { return time_source_; }
 
 private:
   const std::string stats_prefix_;
+  Runtime::Loader& runtime_;
   TimeSource& time_source_;
+  const std::string enabled_runtime_key_;
+  const bool enabled_default_value_;
```

Comment:
Can you move these 2 members into a helper class as well as the `filterEnabled()` function (with a more generic name). More things will end up using this feature flag enabled structure. Please also add dedicated tests for this as well.